src/edn/mod.rs

```diff
@@ -516,7 +517,9 @@ impl std::str::FromStr for Edn {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        crate::deserialize::from_str(s)
+        let tokens = deserialize::tokenize(s);
```

Comment:
didn't understand?

Reply:
I move the code from old `from_str` to `Edn::from_str`, so now it tokenizes, than parse the string to an `Edn` here, just like before.